concepts/contexts.py

```diff
@@ -711,6 +711,25 @@ def shape(self) -> _common.Shape:
             Shape(objects=6, properties=10)
         """
         return _common.Shape._from_pair(self.objects, self.properties)
+    
+    @tools.lazyproperty
+    def density(self) -> float:
```

Comment:
WDYT about `fill_ratio` (used by Outrata, and Vychodil)?

Reply:
Probably more suitable for FCA domain. We can consider `density` alias later (if `fill_ratio` won't be understand by more general domain users).